pkg/controller/certificaterequest/issue_certificate.go

```diff
@@ -53,80 +54,59 @@ func (r *ReconcileCertificateRequest) IssueCertificate(reqLogger logr.Logger, cr
 		reqLogger.Info("operator is configured to use Let's Encrypt staging environment.")
 	}
 
-	letsEncryptClient, err := GetLetsEncryptClient(useLetsEncryptStagingEndpoint)
+	leClient, err := leclient.GetLetsEncryptClient(useLetsEncryptStagingEndpoint)
+	err = leClient.GetAccount(r.client, useLetsEncryptStagingEndpoint, config.OperatorNamespace)
 	if err != nil {
 		return err
 	}
 
-	accountUrl, err := GetLetsEncryptAccountUrl(r.client, useLetsEncryptStagingEndpoint)
-	if err != nil {
-		return err
-	}
-
-	privateKey, err := GetLetsEncryptAccountPrivateKey(r.client, useLetsEncryptStagingEndpoint)
-	if err != nil {
-		return err
-	}
-
-	letsEncryptAccount := acme.Account{PrivateKey: privateKey, URL: accountUrl}
-
-	certExpiryNotificationList := GetCertExpiryNotificationList(cr.Spec.Email)
-
-	letsEncryptAccount, err = letsEncryptClient.UpdateAccount(letsEncryptAccount, true, certExpiryNotificationList...)
+	err = leClient.UpdateAccount(cr.Spec.Email)
 	if err != nil {
 		return err
 	}
 
 	var certDomains []string
-	var ids []acme.Identifier
 
 	for _, domain := range cr.Spec.DnsNames {
-		reqLogger.Info(fmt.Sprintf("%v domain will be added to certificate request", domain))
 		certDomains = append(certDomains, domain)
-		ids = append(ids, acme.Identifier{Type: "dns", Value: domain})
 	}
 
-	letsEncryptOrder, err := letsEncryptClient.NewOrder(letsEncryptAccount, ids)
+	err = leClient.CreateOrder(cr.Spec.DnsNames)
 	if err != nil {
 		return err
 	}
+	URL, _ := leClient.GetOrderURL()
+	reqLogger.Info("created a new order with Let's Encrypt.", "URL", URL)
 
-	reqLogger.Info("created a new order with Let's Encrypt.", "letsEncryptOrder.URL", letsEncryptOrder.URL)
-
-	for _, authUrl := range letsEncryptOrder.Authorizations {
-
-		authorization, err := letsEncryptClient.FetchAuthorization(letsEncryptAccount, authUrl)
+	for _, authURL := range leClient.OrderAuthorization() {
+		err := leClient.FetchAuthorization(authURL)
 		if err != nil {
 			reqLogger.Error(err, "could not fetch authorizations")
 			return err
 		}
 
-		reqLogger.Info("authorization url", "authorization.URL", authorization.URL)
-
-		domain := authorization.Identifier.Value
-
-		challenge, ok := authorization.ChallengeMap[acme.ChallengeTypeDNS01]
-		if !ok {
+		domain := leClient.GetAuthorizationIndentifier()
```

Comment:
Is it possible for this response to be `nil`?

Reply:
Do not believe so. But I'll handling just in case.